google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java

```diff
@@ -729,4 +730,46 @@ public void testDelete() {
     assertNull(keys.next());
     assertFalse(keys.hasNext());
   }
+
+  @Test
+  public void testRunInTransaction() {
+    Datastore.TransactionCallable<Integer> callable1 =
+        new Datastore.TransactionCallable<Integer>() {
+          private Integer attempts = 0;
+
+          public Integer run(DatastoreReaderWriter transaction) {
+            transaction.get(KEY1);
+            if (attempts < 1) {
+              ++attempts;
+              throw new DatastoreException(10, "", "ABORTED", false, null);
+            }
+
+            return attempts;
+          }
+        };
+
+    int result = DATASTORE.runInTransaction(callable1);
+    assertEquals(result, 1);
+
+    Datastore.TransactionCallable<Integer> callable2 =
+        new Datastore.TransactionCallable<Integer>() {
+          private Integer attempts = 0;
+
+          public Integer run(DatastoreReaderWriter transaction) {
+            transaction.get(KEY1);
+            if (attempts < 1) {
+              ++attempts;
+              throw new DatastoreException(4, "", "DEADLINE_EXCEEDED", false, null);
+            }
+            return attempts;
+          }
+        };
+
+    try {
+      DATASTORE.runInTransaction(callable2);
+      fail("Expecting a failure");
+    } catch (DatastoreException expected) {
+      assertEquals(((DatastoreException) expected.getCause()).getCode(), 4);
```

Comment:
This thest succeed also if some other exception (not the DatastoreException) is thrown (and this check is not performed at all). Please change it. There are multiple ways to do it. One possible:
Change method annotation from `@Test` to @`Test(expected = DatastoreException.class) `and then after the line 772: `thow expected;`